impl_core/src/main/java/io/opencensus/implcore/trace/propagation/BinaryFormatImpl.java

```diff
@@ -96,25 +110,38 @@ public SpanContext fromByteArray(byte[] bytes) throws SpanContextParseException
     if (bytes.length == 0 || bytes[0] != VERSION_ID) {
       throw new SpanContextParseException("Unsupported version.");
     }
-    TraceId traceId = TraceId.INVALID;
-    SpanId spanId = SpanId.INVALID;
+    if (bytes.length < FORMAT_LENGTH - ID_SIZE - TraceOptions.SIZE) {
```

Comment:
Maybe use something like:
private static final int REQUIRED_FORMAT_LENGHT = 3 * ID_SIZE + TraceId.SIZE + SpanId.SIZE;
private static final int ALL_FORMAT_LENGHT = REQUIRED_FORMAT_LENGHT + ID_SIZE + TraceOption.SIZE;

Reply:
sgtm doing this now

Reply:
pushed. looks nicer and easier to understand now. good call